### PR TITLE
ADR 009: InfoReservedByte

### PR DESCRIPTION
**Pull request by rootulp**
## Description

Still a work in progress. In particular, I will add more details to the implementation details section

### CODE REVIEW

File: docs/celestia-architecture/adr-009-info-reserved-byte.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+// TODO
+
+## Implementation Details
+
+### Protobuf
+
+1. Add `Version` to [`MsgWirePayForData`](https://github.com/celestiaorg/celestia-app/blob/main/proto/payment/tx.proto#L19)
+1. Add `Version` to [`MsgPayForData`](https://github.com/celestiaorg/celestia-app/blob/main/proto/payment/tx.proto#L44)
+
+**NOTE**: Protobuf does not support the byte type (see [Scalar Value Types](https://developers.google.com/protocol-buffers/docs/proto3#scalar)) so a `uint32` will be used for `Version`. Since `Version` is constrained to 2^7 bits (0 to 127 inclusive), a `Version` outside the supported range (i.e. 128) will seriealize / deserialize correctly but be considered invalid by the application.
```
Comment (evan-forbes) [inline docs/celestia-architecture/adr-009-info-reserved-byte.md:49]: I not sure if its worth putting specifically in the ADR, but for posterity/other reviewers, what the user signs over will only be a single byte + protobuf overhead, as protobuf uses a varint encoding for all number types besides floats and what not.

**Review by evan-forbes — COMMENTED** [review]
nice! the only thing left is to add it when splitting shares! simple pimple. well done

**Comment by evan-forbes** [comment]
also, do we want to move this to the app, considering that's where all the logic will be and this ADR modifies application specific types?

File: docs/celestia-architecture/adr-009-info-reserved-byte.md (hunk 1 above)
Comment (rootulp) [reply docs/celestia-architecture/adr-009-info-reserved-byte.md]: Thanks for sharing!

> as protobuf uses a varint encoding for all number types

Agreed based on https://developers.google.com/protocol-buffers/docs/encoding which states Varint is used for `uint32`.  It's not clear to me how to estimate protobuf overhead so added a sentence in https://github.com/celestiaorg/celestia-core/pull/843/commits/46203433b44d50bbcfc1e6df649da3ffe4a53c76

**Comment by rootulp** [comment]
> nice! the only thing left is to add it when splitting shares! simple pimple. well done

Added in https://github.com/celestiaorg/celestia-core/pull/843/commits/26884d1819fcce6c0d70325d5f017af0e55a7b54 . Resolving https://github.com/celestiaorg/celestia-core/issues/842 and merging https://github.com/celestiaorg/celestia-app/pull/637 would make implementing this ADR significantly easier to tackle.

**Comment by rootulp** [comment]
> also, do we want to move this to the app, considering that's where all the logic will be and this ADR modifies application specific types?

Agreed. I originally targeted celestia-core because all the relevant issues are in this repo. Resolving this ADR will involve modifying celestia-core types too (e.g. `Message` struct) but will move to celestia-app because that's where share logic will live long term.

**Pull request closed without merging**